src/services/mentees.js

```diff
@@ -758,4 +758,162 @@ module.exports = class MenteesHelper {
 			return error
 		}
 	}
+
+	/**
+	 * List mentees and search with name , email
+	 * @method
+	 * @name list
+	 * @param {String} userId - User ID of the mentee.
+	 * @param {Number} pageNo - Page No.
+	 * @param {Number} pageSize - Page Size.
+	 * @param {String} searchText
+	 * @param {String} queryParams
+	 * @param {String} userId
+	 * @param {Boolean} isAMentor - true/false.
+	 * @returns {Promise<Object>} - returns the list of mentees
+	 */
+	static async list(pageNo, pageSize, searchText, queryParams, userId, isAMentor) {
+		try {
+			let additionalProjectionString = ''
+
+			// check for fields query
+			if (queryParams.fields && queryParams.fields !== '') {
+				additionalProjectionString = queryParams.fields
+				delete queryParams.fields
+			}
+			let userServiceQueries = {}
+			let organization_ids = []
+			for (let key in queryParams) {
+				if (queryParams.hasOwnProperty(key) & ((key === 'email') | (key === 'search') | (key === 'name'))) {
+					userServiceQueries[key] = queryParams[key]
+				} else if (queryParams.hasOwnProperty(key) & (key === 'organization_ids')) {
+					organization_ids = queryParams[key].split(',')
+				}
+			}
+
+			const query = utils.processQueryParametersWithExclusions(queryParams)
+
+			let validationData = await entityTypeQueries.findAllEntityTypesAndEntities({
+				status: common.ACTIVE_STATUS,
+			})
+
+			const filteredQuery = utils.validateFilters(query, JSON.parse(JSON.stringify(validationData)), 'sessions')
+
+			const userType = common.MENTEE_ROLE
+
+			const saasFilter = await utils.filterUserListBasedOnSaasPolicy(userId, isAMentor)
+			let extensionDetails = await menteeQueries.getUsersByUserIdsFromView(
+				[],
+				null,
+				null,
+				filteredQuery,
+				saasFilter,
+				additionalProjectionString,
+				true
+			)
+			if (extensionDetails.count == 0) {
+				return common.successResponse({
+					statusCode: httpStatusCode.ok,
+					message: 'MENTEE_LIST',
+					result: {
+						data: [],
+						count: 0,
+					},
+				})
+			}
+			const menteeIds = extensionDetails.data.map((item) => item.user_id)
+
+			if (menteeIds) {
+				userServiceQueries['user_ids'] = menteeIds
+			}
+
+			const userDetails = await userRequests.search(userType, pageNo, pageSize, searchText, userServiceQueries)
```

Comment:
@adithyadinesh0412 make sure all function which is using search API also have these changes